mypy/plugin.py

```diff
@@ -379,3 +421,332 @@ def int_pow_callback(ctx: MethodContext) -> Type:
+                                lhs.is_inferred_def = False
+
+                    # If the attrib has a converter function take the type of the first argument
+                    # as the init type.
+                    converter = get_argument(stmt.rvalue, 'converter')
+                    convert = get_argument(stmt.rvalue, 'convert')
+
+                    if convert and converter:
+                        ctx.api.fail("Can't pass both `convert` and `converter`.", stmt.rvalue)
+                    elif convert:
+                        # Note: Convert is deprecated but works the same.
+                        converter = convert
+
+                    if (converter
+                            and isinstance(converter, RefExpr)
+                            and converter.node
+                            and isinstance(converter.node, FuncBase)
+                            and converter.node.type
+                            and isinstance(converter.node.type, CallableType)
+                            and converter.node.type.arg_types):
+                        typ = converter.node.type.arg_types[0]
+
+                    # Does this even have to go in init?
+                    init = get_bool_argument(stmt.rvalue, 'init', True)
+
+                    if name in attributes:
+                        del attributes[name]
+                    attributes[name] = Attribute(name, typ, attr_has_default, init, stmt)
+                elif auto_attribs and typ and stmt.new_syntax and not is_class_var(lhs):
+                    # `x: int` (without equal sign) assigns rvalue to TempNode(AnyType())
+                    has_rhs = not isinstance(stmt.rvalue, TempNode)
+                    if name in attributes:
+                        del attributes[name]
+                    attributes[name] = Attribute(name, typ, has_rhs, True, stmt)
+
+            elif isinstance(stmt, Decorator):
+                # Look for attr specific decorators.  ('x.default' and 'x.validator')
+                remove_me = []
+                for func_decorator in stmt.decorators:
+                    if (isinstance(func_decorator, MemberExpr)
+                            and isinstance(func_decorator.expr, NameExpr)
+                            and func_decorator.expr.name in attributes):
+
+                        if func_decorator.name == 'default':
+                            # This decorator lets you set a default after the fact.
+                            attributes[func_decorator.expr.name].has_default = True
+
+                        if func_decorator.name in ('default', 'validator'):
+                            # These are decorators on the attrib object that only exist during
+                            # class creation time.  In order to not trigger a type error later we
+                            # just remove them.  This might leave us with a Decorator with no
+                            # decorators (Emperor's new clothes?)
```

Comment:
I think this is fragile. Some code can crash because of this. A proper solution would be to replace it with a ``FuncDef``.

Reply:
I've seen this done in the code base though:  https://github.com/python/mypy/blob/master/mypy/semanal.py#L2921

Reply:
OK, let us hope this will not break other code.